File: data/EXP01/golang_no_system_var/2178/2178.output_1.go
func maximumEvenSplit(finalSum int64) []int64 {
    var result []int64
    divisor := int64(2)

    for finalSum > 0 && divisor <= finalSum {
        if finalSum-divisor != divisor && (finalSum-divisor)%2 == 0 {
            result = append(result, divisor)
            finalSum -= divisor
        }
        divisor += 2
    }

    if finalSum == 0 {
        return result
    }
  
    return []int64{}
}

File: data/EXP01/python/2178/2178.output_1.py
# @lc app=leetcode id=2178 lang=python3
class Solution:
    def maximumEvenSplit(self, finalSum: int) -> List[int]:
        result = []
        largestEven = finalSum - (finalSum % 2)
        
        if largestEven == finalSum:
            result.append(largestEven)
            return result
        
        remainingSum = finalSum - largestEven
        start = 2
        
        while remainingSum > 0:
            if start % 2 == 0 and start <= remainingSum:
                result.append(start)
                remainingSum -= start
            start += 2
        
        if remainingSum == 0:
            return result
        else:
            return []